app/models/ability.rb

```diff
@@ -60,16 +60,28 @@ def collection_abilities(user)
 
   def task_abilities(user)
     can %i[index create import_start import_confirm], Task
-
     alias_action :export_external_start, :export_external_check, :export_external_confirm, to: :export
-    can %i[crud export download add_to_collection], Task, user: {id: user.id}
+    can %i[show export download add_to_collection], Task do |task|
```

Comment:
With this ability, we allow users to add a public task to a collection. Now, I wondered what happens if someone adds a public task to a collection and the task's author later decides to switch the visibility to private. In this scenario, the external user probably should not have access to the task any longer. But I assume that it (and the title) is still shown in the collection? Do we need to add some special handling to this case of changing the visibility to private later on? One option could be to hide all private tasks for collections, so that they are still included if the visibility is turned back to "public". 🤔